src/withAxiom.ts

```diff
@@ -93,24 +96,26 @@ export function withAxiomRouteHandler(handler: NextHandler): NextHandler {
       }
       throw error;
     }
-  };
+  }) as T;
```

Comment:
This is the unsafe cast I mean. 

If you do this, which would be the proper way, there is a type error I can't fix:

```ts
export function withAxiomRouteHandler<T extends AxiomHandler>(handler: T): T {
  const result: T = async (req, arg) => { ... }
  return result
}
```

```log
Type '(req: AxiomRequest, arg: any) => Promise<Response | NextResponse<unknown>>' is not assignable to type 'T'.
  '(req: AxiomRequest, arg: any) => Promise<Response | NextResponse<unknown>>' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint 'AxiomHandler<any, Response | NextResponse<unknown>>'.ts(2322)
```